appveyor.yml

```diff
@@ -1,13 +1,17 @@
 version: '{build}'
 
 environment:
-  mssql_jdbc_test_connection_properties: jdbc:sqlserver://localhost:1433;instanceName=SQL2016;databaseName=master;username=sa;password=Password12!;
+  JAVA_HOME: C:\Program Files\Java\jdk1.8.0 
+
   matrix:
-    - JAVA_HOME: C:\Program Files\Java\jdk1.8.0 
+    - mssql_jdbc_test_connection_properties: jdbc:sqlserver://localhost:1433;instanceName=SQL2008R2SP2;databaseName=master;username=sa;password=Password12!;
+
+    - mssql_jdbc_test_connection_properties: jdbc:sqlserver://localhost:1433;instanceName=SQL2016;databaseName=master;username=sa;password=Password12!;
 
 services:
+  - mssql2008r2sp2
```

Comment:
Will it work? 
I guess you should add services also in matrix. 

````
matrix:
  - SQL: mssql2008r2sp2
   mssql_jdbc_test_connection_properties: jdbc:sqlserver://localhost:1433;instanceName=SQL2008R2SP2;databaseName=master;username=sa;password=Password12!;
  - SQL: mssql2016
   mssql_jdbc_test_connection_properties: jdbc:sqlserver://localhost:1433;instanceName=SQL2016;databaseName=master;username=sa;password=Password12!;
````
I did not check this but definitely you want matrix of services too. 

Ref: https://stackoverflow.com/questions/42690548/multiple-combinations-of-python-and-sql-server-on-appveyor

> Also please note that all MS SQL servers use the same default port 1433. Therefore please start and stop them sequentially to avoid port conflicts. To allow all SQL Server instances to be started simultaneously, please run this script at init stage.

Ref: https://www.appveyor.com/docs/services-databases/